### ISSUE TITLE
at.fast_ring functions need to be cleaned up

### ISSUE THREAD

**Issue by lcarver** [atcollab/at#529]

**Comment by lfarv** [comment]
Agreed. I was about to fully rewrite both python and Matlab versions. It can be made much simpler!

**Comment by lcarver** [comment]
The initial implementation of pyat fast_ring was heavily influenced by the matlab version so the logic is quite similar and not very pythonic. It can be significantly improved.

**Comment by lfarv** [comment]
I have versions rewritten from scratch which are much cleaner !

**Comment by swhite2401** [comment]
For periodic lattices I would suggest to always develop the `ring` before generating the `fastring`

**Comment by swhite2401** [comment]
Done in #1082
